packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -176,10 +200,61 @@ class CodeSnippetDisplay extends React.Component<ICodeSnippetDisplayProps> {
       },
       {
         title: 'Insert',
-        icon: addIcon,
+        icon: importIcon,
         onClick: (): void => {
           this.insertCodeSnippet(codeSnippet);
         }
+      },
+      {
+        title: 'Edit',
+        icon: editIcon,
+        onClick: (): void => {
+          this.props.openCodeSnippetEditor({
+            metadata: [
+              {
+                label: 'Name',
+                value: codeSnippet.displayName,
+                type: 'TextInput',
+                schemaField: 'display_name'
+              },
+              {
+                label: 'Description',
+                value: codeSnippet.description,
+                type: 'TextInput',
+                schemaField: 'description'
+              },
+              {
+                label: 'Language',
+                value: {
+                  choice: codeSnippet.language,
+                  defaultChoices: defaultLanguageChoices
```

Comment:
At first, I was thinking we'd only use languages that are present in the currently configured kernels, but we should allow code snippets in any language since they won't necessarily be tied to a kernel.

Can a user bring their own language and, if so, should the `defaultLanguageChoices` be dynamic to include all languages from _existing_ code snippets that might not be in the hard-coded list?
Since we don't validate the language (against known languages), the server doesn't care what this value is so long as it meets the general criteria of the schema.

Reply:
I'm changing it to check for other languages in existing snippets, but use python, R, java, scala, and markdown as the default language options.